controllers/druid/configuration.go

```diff
@@ -0,0 +1,128 @@
+package druid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func makeConfigMap(name string, namespace string, labels map[string]string, data map[string]string) (*v1.ConfigMap, error) {
+	return &v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Data: data,
+	}, nil
+}
+
+func (r *DruidReconciler) makeCommonConfigMap(ctx context.Context, m *v1alpha1.Druid, ls map[string]string) (*v1.ConfigMap, error) {
```

Comment:
Lets not add methods now. This will need a big change on all the code structure.  Not just configmaps. 
Ideally we would want another interface as ```druidBuilder``` to expose this methods.

Reply:
Removed it from the Reconciler object.  Will fix the e2e and ping you